app/sdil/controllers/ReturnsController.scala

```diff
@@ -197,21 +216,82 @@ class ReturnsController (
+      messages = Map(
+        "pack-at-business-address-in-return.lead" -> s"${Address.fromUkAddress(subscription.address).nonEmptyLines.mkString("<br/>")}")
+    )
+    for {
+      usePPOBAddress   <- ask(bool, "pack-at-business-address-in-return")(implicitly, implicitly, extraMessages)
+      packingSites     = if (usePPOBAddress) {
+        List(Site.fromAddress(Address.fromUkAddress(subscription.address)))
+      } else {
+        List.empty[Site]
+      }
+      firstPackingSite <- ask(packagingSiteMapping,"first-production-site")(packagingSiteForm, implicitly, ExtraMessages()) when
+        packingSites.isEmpty
+      packingSites     <- askPackSites(packingSites ++ firstPackingSite.fold(List.empty[Site])(x => List(x)))
+    } yield packingSites
+  }
+
   private def program(period: ReturnPeriod, subscription: Subscription, sdilRef: String)
                      (implicit hc: HeaderCarrier): WebMonad[Result] = for {
     sdilReturn     <- askReturn(subscription, sdilRef)
+    // check if they need to vary
+    isNewImporter   = (sdilReturn.importLarge |+| sdilReturn.importSmall) > 0 && !subscription.activity.importer
+    isNewPacker     = (sdilReturn.packLarge |+| sdilReturn.packSmall.total) > 0 && !subscription.activity.contractPacker
+    inner           = uniform.fragments.return_variation_continue(isNewImporter, isNewPacker)
+    _               <- tell("return-change-registration", inner) when isNewImporter || isNewPacker
+    newPackingSites <- askNewPackingSites(subscription) when isNewPacker && subscription.productionSites.isEmpty
+    newWarehouses   <- askNewWarehouses when isNewImporter && subscription.warehouseSites.isEmpty
+
+    variation     = ReturnsVariation(
+      orgName = subscription.orgName,
+      ppobAddress = subscription.address,
+      importer = (isNewImporter, (sdilReturn.importLarge |+| sdilReturn.importSmall).combineN(4)),
+      packer = (isNewPacker, (sdilReturn.packLarge |+| sdilReturn.packSmall.total).combineN(4)),
+      warehouses = newWarehouses.getOrElse(List.empty[Site]),
+      packingSites = newPackingSites.getOrElse(List.empty[Site]),
+      phoneNumber = subscription.contact.phoneNumber,
+      email = subscription.contact.email,
+      taxEstimation = taxEstimation(sdilReturn)
+    )
     broughtForward <- BigDecimal("0").pure[WebMonad]
-    _              <- checkYourAnswers("check-your-answers", sdilReturn, broughtForward, subscription.activity.smallProducer)
+    _              <- checkYourAnswers("check-your-answers", sdilReturn, broughtForward, variation, subscription)
     _              <- cachedFuture(s"return-${period.count}")(
                         sdilConnector.returns(subscription.utr, period) = sdilReturn)
+    _              <- if (isNewImporter || isNewPacker) {
+                        execute(sdilConnector.returns.variation(variation, sdilRef))
+                      } else {
+                        (()).pure[WebMonad]
+                      }
```

Comment:
Do you think we need a uniform construct for "if (a) then do b and throw away the result"?

Reply:
Yes please.